core/utils/ethabi.go

```diff
@@ -14,8 +15,23 @@ import (
 )
 
 const (
```

Comment:
I think if somebody without context were to stumble on this section, they'd be completely confused about the differences between the different options. Can we document this more clearly and provide context?

Reply:
Yep, I think you're right. Do you mind if I address this and other issues in a separate PR, so that we can get this merged for @spooktheducks ?

Reply:
That's fine by me, but let's make sure to get it done soon while we still have the context in our mental caches :)

Reply:
I added a bit more of an explanation about the role of these constants, and moved them and their tests into the `adapters` package, to clarify their relationship to `EthTx`